server/routes/connection-schema.js

```diff
@@ -42,8 +43,17 @@ async function getConnectionSchema(req, res) {
   return res.utils.data(schemaInfo);
 }
 
+// compression is added here becasue a big database server can have huge amount
+// of metadata and since this is not retrieved schema by schema 20mb+ would easily be possible in plain/text
+// on slow connections where a LB does not compress this can be a big bottleneck.
+// better would be to expand this router to provide:
+// - '/api/connections/:connectionId/schema_names' to get all schema names
+// - '/api/connections/:connectionId/schema/:schemaName' to get tables of a schema
+// - '/api/connections/:connectionId/schema/:schemaName/:tableName' to get column information of a schema table
+// and then deprecate the current '/api/connections/:connectionId/schema'
```

Comment:
While this would be better for smaller API responses, this would not be compatible with autocomplete unfortunately, as autocomplete requires knowledge of the full schema today. Could you either remove this portion of the comment or add a clarification about autocomplete?

Reply:
Hi for now i have removed the comment from the code since adding it there without proper design was not a good plan, sorry for the confusion. 

My though behind that was not alter the clients ability to load the schema. Only the way it is retrieved, currently it is an all or nothing approach this limits the users feedback on what is happening. I do not mind that all is loaded but it should be done with a lot more user feedback. (e.g. a proper loader)
- 'Retrieving schema names....'
- 'Retrieving tables/columns for schema x/x columns retrieved x' << update as data loading promises are completed.
- Show tree

An other advantage of this staged approach is timeout of the client request is less likely to happen due to long wait or abort due to impatient user refreshing the page.

Also the current setup caches the schema flat this is something that could be more structured. Instead of a plain cache it should be considered to load the entire schema into the backend database of sqlpad. Since each driver could provide a different way of loading the data (and be potentially very time consuming) but every driver always presents it the same way. This would allow for searching and loading of schemas to happen against the backend-database that can be indexed. e.g. a tables structure like:
```
connection_schema_version {
 id,
 connection_id,
 timestamp
}
connection_schema (
 connection_schema_version_id,
 schema_name,
 table_name,
 column_name,
 column_type
)
```
The forced refresh can then load a new version and once completed can delete the old version.